monai/inferers/splitter.py

```diff
@@ -58,142 +91,372 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
         device: torch.device | str | None = None,
-        pad_mode: str | None = PytorchPadMode.CONSTANT,
-        **pad_kwargs: Any,
     ) -> None:
-        """Split the input into patches with sliding window strategy and a possible overlap.
-        If also allows to offset the starting position and filter the patches.
-
-        Args:
-            patch_size : the size of the patch to be generated.
-            offset: the amount of offset for the patches with respect to the original input.  Defaults to 0.
-            overlap: the amount of overlap between patches in each dimension [0, 1). Defaults to 0.0.
-            filter_fn: a callable to filter patches. It should accepts exactly two parameters (patch, location), and
-                return True for a patch to keep. Defaults to no filtering.
-            device: the device where the patches are generated. Defaults to the device of inputs.
-            pad_mode: the pad mode when the patches are extracted from outside of the image
-                (either when the offset is negative or the image is non-divisible by the patch_size).
-                If set to `None`, the last incomplete patch will be dropped. Defaults to PytorchPadMode.CONSTANT.
-            pad_kwargs: other arguments for `torch.nn.functional.pad`.
-
-        Note:
-            If only one scaler value is provided for `patch_size`, `offset`, or `overlap`,
-                it will be broadcasted to all the spatial dimensions.
-        """
         super().__init__(patch_size=patch_size, device=device)
         self.offset = offset
-        if any(ov < 0 or ov >= 1 for ov in ensure_tuple(overlap)):
-            raise ValueError(f"Overlap must be between 0 and 1 but {overlap} is given.")
+        # check if fraction overlaps are within the range of [0, 1)
+        if isinstance(ensure_tuple(overlap)[0], float) and any(ov < 0.0 or ov >= 1.0 for ov in ensure_tuple(overlap)):
+            raise ValueError(
+                f"Relative overlap must be between 0.0 and 1.0 but {overlap} is given. "
+                "If you wish to use number of pixels as overlap, please provide integer numbers."
+            )
+        elif any(ov < 0 for ov in ensure_tuple(overlap)):
+            raise ValueError(f"Number of pixels for overlap cannot be negative. {overlap} is given. ")
+
         self.overlap = overlap
-        self.filter_fn = self._get_valid_filter_fn(filter_fn)
+        self.filter_fn = self._validate_filter_fn(filter_fn)
+        # padding
         self.pad_mode = pad_mode
-        self.pad_kwargs = pad_kwargs
+        self.pad_value = pad_value
+        # check a valid padding mode is provided if there is any negative offset.
+        if not self.pad_mode and any(off < 0 for off in ensure_tuple(offset)):
+            raise ValueError(f"Negative `offset`requires a valid padding mode but `mode` is set to {self.pad_mode}.")
+        # batch and color are non-spatial dimensions
+        self.non_spatial_ndim = 2
```

Comment:
Do you mean "batch and channel" here? And seems we have already assume that by default the input image size is BCHW[D], maybe don't need this temporary variable anymore，what do you think?

Reply:
Sure, I can simplify that too. Thanks

Reply:
Sure, I can change it.